extra/provision.sh

```diff
@@ -12,11 +39,116 @@ DB="fbctf"
+  printf "\nParameters:\n"
+  printf "  -h, --help \t\tShows this help message and exit.\n"
+  printf "  -m MODE, --mode MODE \tMode of operation. Default value is dev\n"
+  printf "  -c TYPE, --cert TYPE \tType of certificate to use. Default value is self\n"
+  printf "\nArguments for MODE:\n"
+  printf "  dev \tProvision will run in development mode. Certificate will be self-signed.\n"
+  printf "  prod \tProvision will run in production mode.\n"
+  printf "\nArguments for TYPE:\n"
+  printf "  self \tProvision will use a self-signed SSL certificate that will be generated.\n"
+  printf "  own \tProvision will use the SSL certificate provided by the user.\n"
+  printf "  cerbot Provision will generate a SSL certificate using letsencrypt/certbot. More info here: https://certbot.eff.org/\n"
+  printf "\nOptional Parameters:\n"
+  printf "  -k PATH, --keyfile PATH \tPath to supplied SSL key file.\n"
+  printf "  -C PATH, --certfile PATH \tPath to supplied SSL certificate pem file.\n"
+  printf "  -D DOMAIN, --domain DOMAIN \tDomain for the SSL certificate to be generated using letsencrypt.\n"
+  printf "  -e EMAIL, --email EMAIL \tDomain for the SSL certificate to be generated using letsencrypt.\n"
+  printf "  -s PATH, --code PATH \t\tPath to fbctf code. Default is /vagrant\n"
+  printf "  -d PATH, --destination PATH \tDestination path to place the fbctf folder. Default is /var/www/fbctf\n"
+  printf "\nExamples:\n"
+  printf "  Provision fbctf in development mode:\n"
+  printf "\t%s -m dev -s /home/foobar/fbctf -d /var/fbctf\n" "${0}"
+  printf "  Provision fbctf in production mode using my own certificate:\n"
+  printf "\t%s -m prod -c own -k /etc/certs/my.key -C /etc/certs/cert.crt -s /home/foobar/fbctf -d /var/fbctf\n" "${0}"
+}
+
+ARGS=$(getopt -n "$0" -o hm:c:k:C:d:m:s:D: -l "help,mode:,cert:,keyfile:,certfile:,domain:,email:,code:,destination:" -- "$@")
+
+eval set -- "$ARGS"
+
+while true; do
+  case "$1" in
+    -h|--help)
+      usage
+      exit 0
+      ;;
+    -m|-mode)
+      GIVEN_ARG=$2
+      if [[ "${VALID_MODE[@]}" =~ "${GIVEN_ARG}" ]]; then
+        MODE=$2
+        shift 2
+      else
+        usage
+        exit 1
+      fi
+      ;;
+    -c|--cert)
+      GIVEN_ARG=$2
+      if [[ "${VALID_TYPE[@]}" =~ "${GIVEN_ARG}" ]]; then
+        TYPE=$2
+        shift 2
+      else
+        usage
+        exit 1
+      fi
+      ;;
+    -k|--keyfile)
+      KEYFILE=$2
+      shift 2
+      ;;
+    -C|--certfile)
+      CERTFILE=$2
+      shift 2
+      ;;
+    -D|--domain)
+      DOMAIN=$2
+      shift 2
+      ;;
+    -e|--email)
+      EMAIL=$2
+      shift 2
+      ;;
+    -s|--code)
+      CODE_PATH=$2
+      shift 2
+      ;;
+    -d|--destination)
+      CTF_PATH=$2
+      shift 2
+      ;;
+    --)
+      shift
+      break
+      ;;
+    *)
+      usage
+      exit 1
+      ;;
+  esac
+done
 
 echo "[+] Provisioning in $MODE mode"
+echo "[+] Using $TYPE certificate"
 
```

Comment:
Can you print out the `code` and `destination` paths here too? It will help when debugging user bug reports.